c_src/filewatch_inotify.c

```diff
@@ -20,38 +29,46 @@ enum { MAX_COOLDOWNS = 12, MSG_LENGTH = 11 };
 */
 struct message {
     int wd;
-    uint32_t len;
+    size_t len;
     char name[NAME_MAX];
 };
 
 struct instance {
     ErlDrvPort port;
     int fd;
-    int len;
-    unsigned n_cooldowns;
-    unsigned long cooldown;
-    size_t pos;
-    struct message msgs[MAX_COOLDOWNS];
-    ErlDrvTermData buf[(MAX_COOLDOWNS * MSG_LENGTH) + 3];
+    size_t n_events;
+    unsigned long cooldown_ms;
+    struct message msgs[MAX_EVENTS];
 };
 
-static void serialize_output(struct instance *self);
-static void serialize_event(struct instance *self, const struct inotify_event *evt);
-static void serialize_message(struct instance *self, struct message *msg);
-static void reset_instance(struct instance *self);
-static void send_output(ErlDrvData self_);
+static bool get_arguments(char *command, unsigned long *cooldown_ms)
+{
+    if (!command || !cooldown_ms) return false;
+
+    command = strchr(command, ' ');
+    if (!command) return false;
 
-static ErlDrvData start(ErlDrvPort port, char *UNUSED)
+    command += strspn(command, " ");
+    if (!*command) return false;
```

Comment:
Might add a quick comment annotating what it is that you're skipping over here. Would avoid having to refer back to the erlang code to figure it out.